text_generation/causal_lm/cpp/generate_pipeline/generate_pipeline.hpp

```diff
@@ -0,0 +1,453 @@
+            std::tie(next_tokens, next_beams) = group_beam_searcher.select_next_tokens(m_model_runner.get_tensor("logits"));
+            if (next_tokens.empty()) {
+                break;
+            }
+            size_t batch_size = next_tokens.size();
+            // Set pointers
+            m_model_runner.set_tensor("input_ids", ov::Tensor{ov::element::i64, {batch_size, 1}, next_tokens.data()});
+            m_model_runner.set_tensor("beam_idx", ov::Tensor{ov::element::i32, {batch_size}, next_beams.data()});
+            // Set auxiliary inputs
+            ov::Tensor attention_mask = m_model_runner.get_tensor("attention_mask");
+            ov::Shape mask_shape{batch_size, attention_mask.get_shape()[1] + 1};
+            attention_mask.set_shape(mask_shape);
+            std::fill_n(attention_mask.data<int64_t>(), ov::shape_size(mask_shape), 1);
+
+            m_model_runner.get_tensor("position_ids").set_shape({batch_size, 1});
+            std::fill_n(m_model_runner.get_tensor("position_ids").data<int64_t>(), batch_size, mask_shape.at(1) - 1);
+            
+            // place
+            sampling_params.m_callback(std::move(next_tokens), *this);
+
+        }
+
+        std::vector<Beam> beams;
+        for (const std::vector<Beam>& group : finalize(std::move(group_beam_searcher))) {
+            for (const Beam& beam : group) {
+                beams.emplace_back(beam);
+            }
+        }
+
+        auto compare_scores = [](Beam left, Beam right) { return (left.score > right.score); };
+        std::sort(beams.begin(), beams.end(), compare_scores);
+        
+        GenerationResult results;
+        for (auto beam = beams.begin(); beam != beams.begin() + sampling_params.m_num_return_sequences; ++beam) {
+            results.emplace_back(std::pair(beam->score, beam->tokens));
+        }
+        return results;
+    }
+
+    GenerationResult multinomial_sampling(ov::Tensor prompts, GenerationConfig sampling_params) {
+        // todo: implement
+        GenerationResult results;
+        return results;
+    }
+
+    std::string call(std::string text) {
+        auto [input_ids, attention_mask] = tokenize(text);
+
+        auto generate_results = generate(input_ids, attention_mask, m_sampling_parameters);
+
+        return detokenize(generate_results)[0];
+    }
+    
+    std::string call(std::string text, GenerationConfig generation_config) {
+        auto [input_ids, attention_mask] = tokenize(text);
+        // to keep config specified during LLMPipeline creation need to get existing 
+        // and modify only after that, e.g.:
+        // GenerationConfig config = pipe.generation_config();
+        // config.do_sample(false).max_new_tokens(20);
+        auto generate_results = generate(input_ids, attention_mask, generation_config);
+
+        return detokenize(generate_results)[0];
+    }
+
+    std::vector<std::string> call(std::vector<std::string> text, GenerationConfig sampling_parameters) {
```

Comment:
do we need `call` method exposed to end users?

Reply:
we don't need that, i made them private